### DISCUSSION TITLE
NPM Overrides not taken into account

### DISCUSSION THREAD

**Discussion by stumueller** [aquasecurity/trivy#4704] [Q&A] [answered]
### Description

There is already an issue for this #3816 However the topic is closed and I can't comment on it.

the new vulnerability in semver <7.5.2 has been flagged by Trivy.
My package.json contains

"overrides": {
    "semver": "^7.5.3"
  },

after doing an npm install my package-lock.json still makes references to semver < 7.5.2 as dependancies.

However an npm audit comes back clean. And an npm list --all shows any semver package as being overridden with 7.5.3
checking npm_modules does not return any semver package apart from 7.5.3

### Desired Behavior

Trivy would take into account the overrides section of package.json

### Actual Behavior

Trivy still reports vulnerabilities

### Reproduction Steps

```bash
1.
2.
3.
...
```

### Target

None

### Scanner

None

### Output Format

None

### Mode

None

### Debug Output

```bash
2023-06-23T15:45:26.982+0100    DEBUG   Severities: ["UNKNOWN" "LOW" "MEDIUM" "HIGH" "CRITICAL"]
2023-06-23T15:45:26.986+0100    DEBUG   cache dir:  /home/stumueller/.cache/trivy
2023-06-23T15:45:26.987+0100    INFO    Need to update DB
2023-06-23T15:45:26.987+0100    INFO    DB Repository: ghcr.io/aquasecurity/trivy-db
2023-06-23T15:45:26.987+0100    INFO    Downloading DB...
37.86 MiB / 37.86 MiB [----------------------------------------------------------------------] 100.00% 7.74 MiB p/s 5.1s2023-06-23T15:45:32.937+0100    DEBUG   Updating database metadata...
2023-06-23T15:45:32.937+0100    DEBUG   DB Schema: 2, UpdatedAt: 2023-06-23 12:10:15.757757128 +0000 UTC, NextUpdate: 2023-06-23 18:10:15.757756628 +0000 UTC, DownloadedAt: 2023-06-23 14:45:32.937556793 +0000 UTC
2023-06-23T15:45:32.937+0100    INFO    Vulnerability scanning is enabled
2023-06-23T15:45:32.937+0100    DEBUG   Vulnerability type:  [os library]
2023-06-23T15:45:32.937+0100    INFO    Secret scanning is enabled
2023-06-23T15:45:32.937+0100    INFO    If your scanning is slow, please try '--scanners vuln' to disable secret scanning
2023-06-23T15:45:32.937+0100    INFO    Please see also https://aquasecurity.github.io/trivy/v0.42/docs/secret/scanning/#recommendation for faster secret detection
2023-06-23T15:45:32.939+0100    DEBUG   No secret config detected: trivy-secret.yaml
2023-06-23T15:45:32.940+0100    DEBUG   Image ID: sha256:5f210ecfbc411743861023aab969fd1961e70f982f32aad89ed04365ec3c3962
2023-06-23T15:45:32.940+0100    DEBUG   Diff IDs: [sha256:78a822fe2a2d2c84f3de4a403188c45f623017d6a4521d23047c9fbb0801794c sha256:fed856459b5efb5ca8ba0ea035f4b3b7cf4954cf35949b4027ae03ab69e22244 sha256:2ed92fd7e053235a103c303aef532cbecd90176c67656610d815e92754e914e7 sha256:34aad8aaa364b41dadc4737ddba82355939a6a027d1cfdf6fd96df14cf602928 sha256:081da77da241640da0c349c066e3438efb1ca1664324ecd4e832238d83a27c00 sha256:edd9e8a27d39c2629fab73076589dccf9147fa1daf6bb831c0a50114481ac1be sha256:2dea5e390911febb3a3f69671c84bb9bed327d5bb1669eaa95bd11f67cc89781]
2023-06-23T15:45:32.940+0100    DEBUG   Base Layers: [sha256:78a822fe2a2d2c84f3de4a403188c45f623017d6a4521d23047c9fbb0801794c sha256:fed856459b5efb5ca8ba0ea035f4b3b7cf4954cf35949b4027ae03ab69e22244 sha256:2ed92fd7e053235a103c303aef532cbecd90176c67656610d815e92754e914e7 sha256:34aad8aaa364b41dadc4737ddba82355939a6a027d1cfdf6fd96df14cf602928]
2023-06-23T15:45:32.946+0100    DEBUG   Missing image ID in cache: sha256:5f210ecfbc411743861023aab969fd1961e70f982f32aad89ed04365ec3c3962
2023-06-23T15:45:32.947+0100    DEBUG   Missing diff ID in cache: sha256:081da77da241640da0c349c066e3438efb1ca1664324ecd4e832238d83a27c00
2023-06-23T15:45:32.947+0100    DEBUG   Missing diff ID in cache: sha256:2dea5e390911febb3a3f69671c84bb9bed327d5bb1669eaa95bd11f67cc89781
2023-06-23T15:45:32.947+0100    DEBUG   Missing diff ID in cache: sha256:edd9e8a27d39c2629fab73076589dccf9147fa1daf6bb831c0a50114481ac1be
2023-06-23T15:45:38.033+0100    DEBUG   No secrets found in container image config
2023-06-23T15:45:38.036+0100    INFO    Detected OS: alpine
2023-06-23T15:45:38.036+0100    INFO    Detecting Alpine vulnerabilities...
2023-06-23T15:45:38.036+0100    DEBUG   alpine: os version: 3.18
2023-06-23T15:45:38.036+0100    DEBUG   alpine: package repository: 3.18
2023-06-23T15:45:38.036+0100    DEBUG   alpine: the number of packages: 17
2023-06-23T15:45:38.037+0100    INFO    Number of language-specific files: 1
2023-06-23T15:45:38.037+0100    INFO    Detecting node-pkg vulnerabilities...
2023-06-23T15:45:38.037+0100    DEBUG   Detecting library vulnerabilities, type: node-pkg, path:
2023-06-23T15:45:38.048+0100    DEBUG   Found an ignore file .trivyignore
2023-06-23T15:45:38.048+0100    DEBUG   These IDs will be ignored: []
2023-06-23T15:45:38.048+0100    DEBUG   Found an ignore file .trivyignore
2023-06-23T15:45:38.048+0100    DEBUG   These IDs will be ignored: []

web (alpine 3.18.2)

Total: 0 (UNKNOWN: 0, LOW: 0, MEDIUM: 0, HIGH: 0, CRITICAL: 0)

2023-06-23T15:45:38.048+0100    INFO    Table result includes only package filenames. Use '--format json' option to get the full path to the package file.

Node.js (node-pkg)

Total: 1 (UNKNOWN: 0, LOW: 0, MEDIUM: 1, HIGH: 0, CRITICAL: 0)

┌───────────────────────┬────────────────┬──────────┬───────────────────┬───────────────┬───────────────────────────────────────────────────────────┐
│        Library        │ Vulnerability  │ Severity │ Installed Version │ Fixed Version │                           Title                           │
├───────────────────────┼────────────────┼──────────┼───────────────────┼───────────────┼───────────────────────────────────────────────────────────┤
│ semver (package.json) │ CVE-2022-25883 │ MEDIUM   │ 7.3.7             │ 7.5.2         │ semver vulnerable to Regular Expression Denial of Service │
│                       │                │          │                   │               │ https://avd.aquasec.com/nvd/cve-2022-25883                │
└───────────────────────┴────────────────┴──────────┴───────────────────┴───────────────┴───────────────────────────────────────────────────────────┘
```

### Operating System

Ubuntu 22.4

### Version

```bash
0.42.1
```

### Checklist

- [ ] Run `trivy --reset`
- [ ] Read [the troubleshooting](https://aquasecurity.github.io/trivy/latest/docs/references/troubleshooting/)

**Comment by knqyf263** [comment]
@DmitriyLewen Can you please take a look?

**Comment by DmitriyLewen** [comment, marked as answer]
Hello @stumueller 
Thanks for your report!

Can you rescan your image in `json` format (`--format json`).
And check file from `PkgPath` field.

Some example from my PC:
```
➜ trivy -d rootfs -f json .        
...
{
  "SchemaVersion": 2,
  "ArtifactName": ".",
  "ArtifactType": "filesystem",
  "Metadata": {
    "ImageConfig": {
      "architecture": "",
      "created": "0001-01-01T00:00:00Z",
      "os": "",
      "rootfs": {
        "type": "",
        "diff_ids": null
      },
      "config": {}
    }
  },
  "Results": [
    {
      "Target": "Node.js",
      "Class": "lang-pkgs",
      "Type": "node-pkg",
      "Vulnerabilities": [
        {
          "VulnerabilityID": "CVE-2022-25883",
          "PkgID": "semver@7.3.7",
          "PkgName": "semver",
          "PkgPath": "node_modules/semver/package.json",
          "InstalledVersion": "7.3.7",
          "FixedVersion": "7.5.2",
          "Layer": {},
          "SeveritySource": "ghsa",
          "PrimaryURL": "https://avd.aquasec.com/nvd/cve-2022-25883",
          "DataSource": {
            "ID": "ghsa",
            "Name": "GitHub Security Advisory Npm",
            "URL": "https://github.com/advisories?query=type%3Areviewed+ecosystem%3Anpm"
          },
          "Title": "semver vulnerable to Regular Expression Denial of Service",
          "Description": "Versions of the package semver before 7.5.2 are vulnerable to Regular Expression Denial of Service (ReDoS) via the function new Range, when untrusted user data is provided as a range.\r\r\r",
          "Severity": "MEDIUM",
          "CVSS": {
            "ghsa": {
              "V3Vector": "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:L",
              "V3Score": 5.3
            }
          },
          "References": [
           ...
          ],
          "PublishedDate": "2023-06-21T05:15:00Z",
          "LastModifiedDate": "2023-06-21T12:29:00Z"
        }
      ]
    }
  ]
}

➜ cat node_modules/semver/package.json
{
  "name": "semver",
  "version": "7.3.7",
  "description": "The semantic version parser used by npm.",
  ...
}

```

**Reply by stumueller** [reply]
@DmitriyLewen here are the results of the scan in full. I'll also comment with my package.json file:
{
  "SchemaVersion": 2,
  "ArtifactName": "web",
  "ArtifactType": "container_image",
  "Metadata": {
    "OS": {
      "Family": "alpine",
      "Name": "3.18.2"
    },
    "ImageID": "sha256:96670e6291d8ec8062f0eb07df1818eddee5e414350829e8df9416dd7d214fef",
    "DiffIDs": [
      "sha256:78a822fe2a2d2c84f3de4a403188c45f623017d6a4521d23047c9fbb0801794c",
      "sha256:fed856459b5efb5ca8ba0ea035f4b3b7cf4954cf35949b4027ae03ab69e22244",
      "sha256:2ed92fd7e053235a103c303aef532cbecd90176c67656610d815e92754e914e7",
      "sha256:34aad8aaa364b41dadc4737ddba82355939a6a027d1cfdf6fd96df14cf602928",
      "sha256:cf0a26dbff758ec585b32ea35ee59823f16938e248d36569fba9c37f0ceb5988",
      "sha256:0084c525ef5e559968dd326ec2427a4f3fd7d3aa1067e6501a1913792a06fce9",
      "sha256:5eeef4c118c6b1a8ba7da27f9f2245786de3f239d7b325eee63a0a27d960659d"
    ],
    "RepoTags": [
      "web:latest"
    ],
    "ImageConfig": {
      "architecture": "amd64",
      "created": "2023-06-27T08:36:56.417223622+01:00",
      "history": [
        {
          "created": "2023-06-14T20:41:58Z",
          "created_by": "/bin/sh -c #(nop) ADD file:1da756d12551a0e3e793e02ef87432d69d4968937bd11bed0af215db19dd94cd in / "
        },
        {
          "created": "2023-06-14T20:41:59Z",
          "created_by": "/bin/sh -c #(nop)  CMD [\"/bin/sh\"]",
          "empty_layer": true
        },
        {
          "created": "2023-06-15T00:26:09Z",
          "created_by": "/bin/sh -c #(nop)  ENV NODE_VERSION=16.20.0",
          "empty_layer": true
        },
        {
          "created": "2023-06-15T00:26:16Z",
          "created_by": "/bin/sh -c addgroup -g 1000 node     \u0026\u0026 adduser -u 1000 -G node -s /bin/sh -D node     \u0026\u0026 apk add --no-cache         libstdc++     \u0026\u0026 apk add --no-cache --virtual .build-deps         curl     \u0026\u0026 ARCH= \u0026\u0026 alpineArch=\"$(apk --print-arch)\"       \u0026\u0026 case \"${alpineArch##*-}\" in         x86_64)           ARCH='x64'           CHECKSUM=\"b3f80fe7d0f1af6fe25ffedc7237ca519965d08fc800eab29cf45cd5b90cdb26\"           ;;         *) ;;       esac   \u0026\u0026 if [ -n \"${CHECKSUM}\" ]; then     set -eu;     curl -fsSLO --compressed \"https://unofficial-builds.nodejs.org/download/release/v$NODE_VERSION/node-v$NODE_VERSION-linux-$ARCH-musl.tar.xz\";     echo \"$CHECKSUM  node-v$NODE_VERSION-linux-$ARCH-musl.tar.xz\" | sha256sum -c -       \u0026\u0026 tar -xJf \"node-v$NODE_VERSION-linux-$ARCH-musl.tar.xz\" -C /usr/local --strip-components=1 --no-same-owner       \u0026\u0026 ln -s /usr/local/bin/node /usr/local/bin/nodejs;   else     echo \"Building from source\"     \u0026\u0026 apk add --no-cache --virtual .build-deps-full         binutils-gold         g++         gcc         gnupg         libgcc
linux-headers         make         python3     \u0026\u0026 export GNUPGHOME=\"$(mktemp -d)\"     \u0026\u0026 for key in       4ED778F539E3634C779C87C6D7062848A1AB005C       141F07595B7B3FFE74309A937405533BE57C7D57       74F12602B6F1C4E913FAA37AD3A89613643B6201       DD792F5973C6DE52C432CBDAC77ABFA00DDBF2B7       61FC681DFB92A079F1685E77973F295594EC4689       8FCCA13FEF1D0C2E91008E09770F7A9A5AE15600       C4F0DFFF4E8C1A8236409D08E73BC641CC11F4C8       890C08DB8579162FEE0DF9DB8BEAB4DFCF555EF4       C82FA3AE1CBEDC6BE46B9360C43CEC45C17AB93C       108F52B48DB57BB0CC439B2997B01419BD92F80A     ; do       gpg --batch --keyserver hkps://keys.openpgp.org --recv-keys \"$key\" ||       gpg --batch --keyserver keyserver.ubuntu.com --recv-keys \"$key\" ;     done     \u0026\u0026 curl -fsSLO --compressed \"https://nodejs.org/dist/v$NODE_VERSION/node-v$NODE_VERSION.tar.xz\"     \u0026\u0026 curl -fsSLO --compressed \"https://nodejs.org/dist/v$NODE_VERSION/SHASUMS256.txt.asc\"     \u0026\u0026 gpg --batch --decrypt --output SHASUMS256.txt SHASUMS256.txt.asc     \u0026\u0026 gpgconf --kill all     \u0026\u0026 rm -rf \"$GNUPGHOME\"     \u0026\u0026 grep \" node-v$NODE_VERSION.tar.xz\\$\" SHASUMS256.txt | sha256sum -c -     \u0026\u0026 tar -xf \"node-v$NODE_VERSION.tar.xz\"     \u0026\u0026 cd \"node-v$NODE_VERSION\"     \u0026\u0026 ./configure     \u0026\u0026 make -j$(getconf _NPROCESSORS_ONLN) V=     \u0026\u0026 make install     \u0026\u0026 apk del .build-deps-full     \u0026\u0026 cd ..     \u0026\u0026 rm -Rf \"node-v$NODE_VERSION\"     \u0026\u0026 rm \"node-v$NODE_VERSION.tar.xz\" SHASUMS256.txt.asc SHASUMS256.txt;   fi   \u0026\u0026 rm -f \"node-v$NODE_VERSION-linux-$ARCH-musl.tar.xz\"   \u0026\u0026 apk del .build-deps   \u0026\u0026 node --version   \u0026\u0026 npm --version"
        },
        {
          "created": "2023-06-15T00:26:16Z",
          "created_by": "/bin/sh -c #(nop)  ENV YARN_VERSION=1.22.19",
          "empty_layer": true
        },
        {
          "created": "2023-06-15T00:26:20Z",
          "created_by": "/bin/sh -c apk add --no-cache --virtual .build-deps-yarn curl gnupg tar   \u0026\u0026 export GNUPGHOME=\"$(mktemp -d)\"   \u0026\u0026 for key in     6A010C5166006599AA17F08146C2130DFD2497F5   ; do     gpg --batch --keyserver hkps://keys.openpgp.org --recv-keys \"$key\" ||     gpg --batch --keyserver keyserver.ubuntu.com --recv-keys \"$key\" ;   done   \u0026\u0026 curl -fsSLO --compressed \"https://yarnpkg.com/downloads/$YARN_VERSION/yarn-v$YARN_VERSION.tar.gz\"   \u0026\u0026 curl -fsSLO --compressed \"https://yarnpkg.com/downloads/$YARN_VERSION/yarn-v$YARN_VERSION.tar.gz.asc\"   \u0026\u0026 gpg --batch --verify yarn-v$YARN_VERSION.tar.gz.asc yarn-v$YARN_VERSION.tar.gz   \u0026\u0026 gpgconf --kill all   \u0026\u0026 rm -rf \"$GNUPGHOME\"   \u0026\u0026 mkdir -p /opt   \u0026\u0026 tar -xzf yarn-v$YARN_VERSION.tar.gz -C /opt/   \u0026\u0026 ln -s /opt/yarn-v$YARN_VERSION/bin/yarn /usr/local/bin/yarn   \u0026\u0026 ln -s /opt/yarn-v$YARN_VERSION/bin/yarnpkg /usr/local/bin/yarnpkg   \u0026\u0026 rm yarn-v$YARN_VERSION.tar.gz.asc yarn-v$YARN_VERSION.tar.gz   \u0026\u0026 apk del .build-deps-yarn   \u0026\u0026 yarn --version"
        },
        {
          "created": "2023-06-15T00:26:20Z",
          "created_by": "/bin/sh -c #(nop) COPY file:4d192565a7220e135cab6c77fbc1c73211b69f3d9fb37e62857b2c6eb9363d51 in /usr/local/bin/ "
        },
        {
          "created": "2023-06-15T00:26:20Z",
          "created_by": "/bin/sh -c #(nop)  ENTRYPOINT [\"docker-entrypoint.sh\"]",
          "empty_layer": true
        },
        {
          "created": "2023-06-15T00:26:20Z",
          "created_by": "/bin/sh -c #(nop)  CMD [\"node\"]",
          "empty_layer": true
        },
        {
          "created": "2023-06-27T07:36:13Z",
          "created_by": "WORKDIR /app",
          "comment": "buildkit.dockerfile.v0"
        },
        {
          "created": "2023-06-27T07:36:53Z",
          "created_by": "COPY /build/build public/ # buildkit",
          "comment": "buildkit.dockerfile.v0"
        },
        {
          "created": "2023-06-27T07:36:56Z",
          "created_by": "RUN /bin/sh -c npm i -g http-server # buildkit",
          "comment": "buildkit.dockerfile.v0"
        },
        {
          "created": "2023-06-27T07:36:56Z",
          "created_by": "ENTRYPOINT [\"http-server\"]",
          "comment": "buildkit.dockerfile.v0",
          "empty_layer": true
        }
      ],
      "os": "linux",
      "rootfs": {
        "type": "layers",
        "diff_ids": [
          "sha256:78a822fe2a2d2c84f3de4a403188c45f623017d6a4521d23047c9fbb0801794c",
          "sha256:fed856459b5efb5ca8ba0ea035f4b3b7cf4954cf35949b4027ae03ab69e22244",
          "sha256:2ed92fd7e053235a103c303aef532cbecd90176c67656610d815e92754e914e7",
          "sha256:34aad8aaa364b41dadc4737ddba82355939a6a027d1cfdf6fd96df14cf602928",
          "sha256:cf0a26dbff758ec585b32ea35ee59823f16938e248d36569fba9c37f0ceb5988",
          "sha256:0084c525ef5e559968dd326ec2427a4f3fd7d3aa1067e6501a1913792a06fce9",
          "sha256:5eeef4c118c6b1a8ba7da27f9f2245786de3f239d7b325eee63a0a27d960659d"
        ]
      },
      "config": {
        "Entrypoint": [
          "http-server"
        ],
        "Env": [
          "PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",
          "NODE_VERSION=16.20.0",
          "YARN_VERSION=1.22.19"
        ],
        "WorkingDir": "/app"
      }
    }
  },
  "Results": [
    {
      "Target": "web (alpine 3.18.2)",
      "Class": "os-pkgs",
      "Type": "alpine"
    },
    {
      "Target": "Node.js",
      "Class": "lang-pkgs",
      "Type": "node-pkg",
      "Vulnerabilities": [
        {
          "VulnerabilityID": "CVE-2022-25883",
          "PkgID": "semver@7.3.7",
          "PkgName": "semver",
          "PkgPath": "usr/local/lib/node_modules/npm/node_modules/semver/package.json",
          "InstalledVersion": "7.3.7",
          "FixedVersion": "7.5.2",
          "Layer": {
            "DiffID": "sha256:fed856459b5efb5ca8ba0ea035f4b3b7cf4954cf35949b4027ae03ab69e22244"
          },
          "SeveritySource": "ghsa",
          "PrimaryURL": "https://avd.aquasec.com/nvd/cve-2022-25883",
          "DataSource": {
            "ID": "ghsa",
            "Name": "GitHub Security Advisory Npm",
            "URL": "https://github.com/advisories?query=type%3Areviewed+ecosystem%3Anpm"
          },
          "Title": "semver vulnerable to Regular Expression Denial of Service",
          "Description": "Versions of the package semver before 7.5.2 are vulnerable to Regular Expression Denial of Service (ReDoS) via the function new Range, when untrusted user data is provided as a range.\r\r\r",
          "Severity": "MEDIUM",
          "CVSS": {
            "ghsa": {
              "V3Vector": "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:L",
              "V3Score": 5.3
            }
          },
          "References": [
            "https://github.com/advisories/GHSA-c2qf-rxjj-qqgw",
            "https://github.com/npm/node-semver/blob/main/classes/range.js#L97-L104",
            "https://github.com/npm/node-semver/blob/main/classes/range.js%23L97-L104",
            "https://github.com/npm/node-semver/blob/main/internal/re.js#L138",
            "https://github.com/npm/node-semver/blob/main/internal/re.js#L160",
            "https://github.com/npm/node-semver/blob/main/internal/re.js%23L138",
            "https://github.com/npm/node-semver/blob/main/internal/re.js%23L160",
            "https://github.com/npm/node-semver/commit/717534ee353682f3bcf33e60a8af4292626d4441",
            "https://github.com/npm/node-semver/pull/564",
            "https://nvd.nist.gov/vuln/detail/CVE-2022-25883",
            "https://security.snyk.io/vuln/SNYK-JS-SEMVER-3247795"
          ],
          "PublishedDate": "2023-06-21T05:15:00Z",
          "LastModifiedDate": "2023-06-21T12:29:00Z"
        }
      ]
    }
  ]
}

**Comment by stumueller** [comment]
package.json contents:
{
  "name": "mats-frontend-web",
  "version": "1.0.0",
  "description": "Web Frontend UI",
  "dependencies": {
    "@essnextgen/ui-kit": "^0.7.3-alpha.55",
    "@microsoft/applicationinsights-react-js": "^3.3.6",
    "@microsoft/applicationinsights-web": "^2.8.5",
    "@testing-library/jest-dom": "^5.16.4",
    "@testing-library/react": "^11.2.7",
    "@testing-library/react-hooks": "^7.0.2",
    "@testing-library/user-event": "^12.8.3",
    "@types/react-dom": "^17.0.17",
    "axios": "^0.27.2",
    "crypto-js": "^4.1.1",
    "history": "^5.0.1",
    "jest-when": "^3.5.1",
    "js-cookie": "^3.0.1",
    "powerbi-client": "^2.21.1",
    "react": "^17.0.2",
    "react-device-detect": "^1.17.0",
    "react-dom": "^17.0.2",
    "react-error-boundary": "^3.1.4",
    "react-ga4": "^1.4.1",
    "react-gtm-module": "^2.0.11",
    "react-router-dom": "^5.3.3",
    "typescript": "^4.7.4",
    "web-vitals": "^2.1.4"
  },
  "overrides": {
    "semver": "^7.5.3"
  },
  "scripts": {
    "start": "react-scripts start",
    "build": "react-scripts build",
    "test": "react-scripts test",
    "eject": "react-scripts eject",
    "prepare": "husky install",
    "test:ci": "CI=true react-scripts test --testResultsProcessor ./node_modules/jest-junit"
  },
  "repository": {
    "type": "git",
    "url": "git@ssh.dev.azure.com:v3/essnextgen/MAT%20Dashboards/mats-frontend-web"
  },
  "browserslist": {
    "production": [
      ">0.2%",
      "not dead",
      "not op_mini all"
    ],
    "development": [
      "last 1 chrome version",
      "last 1 firefox version",
      "last 1 safari version"
    ]
  },
  "devDependencies": {
    "@types/crypto-js": "^4.1.1",
    "@types/jest": "^26.0.24",
    "@types/jest-when": "^3.5.2",
    "@types/js-cookie": "^2.2.7",
    "@types/mocha": "^9.1.1",
    "@types/node": "^12.20.55",
    "@types/react": "^17.0.48",
    "@types/react-gtm-module": "^2.0.1",
    "@types/react-router-dom": "^5.3.0",
    "@typescript-eslint/eslint-plugin": "^4.31.1",
    "@typescript-eslint/parser": "^4.31.1",
    "eslint": "^7.32.0",
    "eslint-config-prettier": "^8.3.0",
    "eslint-import-resolver-typescript": "^2.7.1",
    "eslint-plugin-flowtype": "^6.0.1",
    "eslint-plugin-import": "^2.26.0",
    "eslint-plugin-jest": "^24.4.2",
    "eslint-plugin-jest-dom": "^4.0.2",
    "eslint-plugin-jsx-a11y": "^6.6.1",
    "eslint-plugin-react": "^7.30.1",
    "eslint-plugin-react-hooks": "^4.6.0",
    "eslint-plugin-testing-library": "^4.12.3",
    "husky": "^8.0.3",
    "jest-junit": "^13.2.0",
    "lint-staged": "^13.2.1",
    "msw": "^0.44.2",
    "node-talisman": "^1.29.10",
    "prettier": "^2.7.1",
    "react-scripts": "^5.0.0",
    "react-test-renderer": "^17.0.2"
  }
}

**Reply by DmitriyLewen** [reply]
> package.json contents:

This is your main `package.json`, right?

Can you send me `usr/local/lib/node_modules/npm/node_modules/semver/package.json` file?

**Reply by stumueller** [reply]
yes that was my main packages.json.

I don't have node_modules under /usr/local/lib our node_modules are in each repo, here is the semver package.json for the repo I'm getting the false positive in. (I've done a search of this node_modules folder and this is the only instance of semver I can find)
{
  "name": "semver",
  "version": "7.5.3",
  "description": "The semantic version parser used by npm.",
  "main": "index.js",
  "scripts": {
    "test": "tap",
    "snap": "tap",
    "lint": "eslint \"**/*.js\"",
    "postlint": "template-oss-check",
    "lintfix": "npm run lint -- --fix",
    "posttest": "npm run lint",
    "template-oss-apply": "template-oss-apply --force"
  },
  "devDependencies": {
    "@npmcli/eslint-config": "^4.0.0",
    "@npmcli/template-oss": "4.15.1",
    "tap": "^16.0.0"
  },
  "license": "ISC",
  "repository": {
    "type": "git",
    "url": "https://github.com/npm/node-semver.git"
  },
  "bin": {
    "semver": "bin/semver.js"
  },
  "files": [
    "bin/",
    "lib/",
    "classes/",
    "functions/",
    "internal/",
    "ranges/",
    "index.js",
    "preload.js",
    "range.bnf"
  ],
  "tap": {
    "timeout": 30,
    "coverage-map": "map.js",
    "nyc-arg": [
      "--exclude",
      "tap-snapshots/**"
    ]
  },
  "engines": {
    "node": ">=10"
  },
  "dependencies": {
    "lru-cache": "^6.0.0"
  },
  "author": "GitHub Inc.",
  "templateOSS": {
    "//@npmcli/template-oss": "This file is partially managed by @npmcli/template-oss. Edits may be overwritten.",
    "version": "4.15.1",
    "engines": ">=10",
    "ciVersions": [
      "10.0.0",
      "10.x",
      "12.x",
      "14.x",
      "16.x",
      "18.x"
    ],
    "npmSpec": "8",
    "distPaths": [
      "classes/",
      "functions/",
      "internal/",
      "ranges/",
      "index.js",
      "preload.js",
      "range.bnf"
    ],
    "allowPaths": [
      "/classes/",
      "/functions/",
      "/internal/",
      "/ranges/",
      "/index.js",
      "/preload.js",
      "/range.bnf"
    ],
    "publish": "true"
  }
}

**Reply by DmitriyLewen** [reply]
> I don't have node_modules under /usr/local/lib

It is strange case, because Trivy found vulnerability for `usr/local/lib/node_modules/npm/node_modules/semver/package.json` file.

Can you upload and share your image (you can remove all source code so you don't share your private work( save only package.json files(from app and from node_modules folder))?

**Reply by stumueller** [reply]
ah I've just thought, was I being daft. The path trivy would be reporting on would be the path in hte image, not in my repo. let me check the package.json in my image and send you that

**Reply by stumueller** [reply]
ok so this is the package.json from the image. And it does specify an insecure version
{
  "name": "semver",
  "version": "7.3.7",
  "description": "The semantic version parser used by npm.",
  "main": "index.js",
  "scripts": {
    "test": "tap",
    "snap": "tap",
    "preversion": "npm test",
    "postversion": "npm publish",
    "postpublish": "git push origin --follow-tags",
    "lint": "eslint \"**/*.js\"",
    "postlint": "template-oss-check",
    "lintfix": "npm run lint -- --fix",
    "prepublishOnly": "git push origin --follow-tags",
    "posttest": "npm run lint",
    "template-oss-apply": "template-oss-apply --force"
  },
  "devDependencies": {
    "@npmcli/eslint-config": "^3.0.1",
    "@npmcli/template-oss": "3.3.2",
    "tap": "^16.0.0"
  },
  "license": "ISC",
  "repository": {
    "type": "git",
    "url": "https://github.com/npm/node-semver.git"
  },
  "bin": {
    "semver": "bin/semver.js"
  },
  "files": [
    "bin/",
    "classes/",
    "functions/",
    "internal/",
    "ranges/",
    "index.js",
    "preload.js",
    "range.bnf"
  ],
  "tap": {
    "check-coverage": true,
    "coverage-map": "map.js"
  },
  "engines": {
    "node": ">=10"
  },
  "dependencies": {
    "lru-cache": "^6.0.0"
  },
  "author": "GitHub Inc.",
  "templateOSS": {
    "//@npmcli/template-oss": "This file is partially managed by @npmcli/template-oss. Edits may be overwritten.",
    "version": "3.3.2",
    "engines": ">=10",
    "ciVersions": [
      "10.0.0",
      "10.x",
      "12.x",
      "14.x",
      "16.x"
    ],
    "distPaths": [
      "bin/",
      "classes/",
      "functions/",
      "internal/",
      "ranges/",
      "index.js",
      "preload.js",
      "range.bnf"
    ]
  }
}

**Reply by DmitriyLewen** [reply]
right! Looks like semver@7.3.7 installed as global package. 
You need to remove it or update.

**Reply by stumueller** [reply]
so the docker build appears to be ignoring the override.
if I delete the node_modules in my local repo and do an nm ci it installs the right version of semver. it doesn't appear to with my docker build tho.

thanks for confirming though
